examples/kms/vault/kms-config.yaml

```diff
@@ -9,6 +9,8 @@ data:
         "vaultAddress": "http://vault.default.svc.cluster.local:8200",
         "vaultAuthPath": "/v1/auth/kubernetes/login",
         "vaultRole": "csi-kubernetes",
+        "vaultTokenAuth": "true",
```

Comment:
we need to validate both token-based and kubernetes secret based encryption. with this change, we are only validating the token-based approach, not the kubernetes secret based.

Reply:
This is bit tricky and inviting thoughts. At present, in our general framework umbrella we deploy vault: 

```
var _ = Describe("RBD", func() {
	f := framework.NewDefaultFramework("rbd")
	var c clientset.Interface
	// deploy RBD CSI
	BeforeEach(func() {
		if !testRBD || upgradeTesting {
			Skip("Skipping RBD E2E")
		}
		c = f.ClientSet
		if deployRBD {
			err := createNodeLabel(f, nodeRegionLabel, regionValue)
			if err != nil {
				e2elog.Failf("failed to create node label with error %v", err)
			}
			err = createNodeLabel(f, nodeZoneLabel, zoneValue)
			if err != nil {
				e2elog.Failf("failed to create node label with error %v", err)
			}
			if cephCSINamespace != defaultNs {
				err = createNamespace(c, cephCSINamespace)
				if err != nil {
					e2elog.Failf("failed to create namespace with error %v", err)
				}
			}
			deployRBDPlugin()
		}
		err := createConfigMap(rbdDirPath, f.ClientSet, f)
		if err != nil {
			e2elog.Failf("failed to create configmap with error %v", err)
		}
		err = createRBDStorageClass(f.ClientSet, f, nil, nil)
		if err != nil {
			e2elog.Failf("failed to create storageclass with error %v", err)
		}
		err = createRBDSecret(f.ClientSet, f)
		if err != nil {
			e2elog.Failf("failed to create secret with error %v", err)
		}
		deployVault(f.ClientSet, deployTimeout) ----------------> 1
	})
```

[1] This can be tweaked to test either token based or other based on the input of configmap or based on the manipulation we could do in deploy vault function. But its single run of one type of vault deployment. We could do the same or repeat it to validate both. Otherwise we may have to take out `deployVault` from general umbrella and try to accommodate both kind of deployment as a first function, then the question comes which tests have to bound to these different invocations? all or just some of them?

Any thoughts here @Madhu-1

Reply:
We need to keep `vaultTokenAuth` as `false` and move the e2e to a helper function and validate the kube auth first and update the configmap with `vaultTokenAuth` = `true` and call the e2e again, validate the token based.